robocasa/environments/kitchen/kitchen.py

```diff
@@ -356,6 +355,66 @@ def _create_obj(cfg):
             return
         self.object_placements = object_placements
 
+        # Helper function to add noises to randomize the robot model.
+        def _generate_random_pos_and_ori(base_xpos, base_ori, randomness_radius):
+            noisy_pos = np.copy(base_xpos)
+            noisy_ori = np.copy(base_ori)
+
+            theta = random.uniform(0, 2 * math.pi)
+            sampled_r = randomness_radius * math.sqrt(random.uniform(0, 1))
+            noise_x = sampled_r * math.cos(theta)
+            noise_y = sampled_r * math.sin(theta)
+            noisy_pos[0] += noise_x
+            noisy_pos[1] += noise_y
+            noisy_ori[2] += random.uniform(0, 2 * math.pi)
+            return noisy_pos, noisy_ori
+
+        robot_base_pos, robot_base_ori = self.compute_robot_base_placement_pose(ref_fixture=ref_fixture)
+        robot_model = self.robots[0].robot_model
+        
+        self.robot_randomness_radius = 0.5
+        if self.robot_randomness_radius is not None:
+            base_fixtures = self._get_base_fixtures()
+            for attempt in range(50):
+                is_valid_pos = True
+                noisy_pos, noisy_ori = _generate_random_pos_and_ori(robot_base_pos, robot_base_ori, self.robot_randomness_radius)
+                
+                # Check all objects first 
+                for (x, y, z), other_quat, other_obj in self.object_placements.values():
+                    if OU.objs_intersect(
+                        obj=robot_model,
```

Comment:
This way of checking `robot_model` is not properly checking its mount. This seems to be an internal issue, but will keep investigating.